\documentclass[author-year, review, 11pt]{components/elsarticle} %review=doublespace preprint=single 5p=2 column
%%% Begin My package additions %%%%%%%%%%%%%%%%%%%
\usepackage[hyphens]{url}
\usepackage{lineno} % add 
  \linenumbers % turns line numbering on 
\bibliographystyle{elsarticle-harv}
\biboptions{sort&compress} % For natbib
\usepackage{graphicx}
\usepackage{booktabs} % book-quality tables
%% Redefines the elsarticle footer
\makeatletter
\def\ps@pprintTitle{%
 \let\@oddhead\@empty
 \let\@evenhead\@empty
 \def\@oddfoot{\it \hfill\today}%
 \let\@evenfoot\@oddfoot}
\makeatother

% A modified page layout
\textwidth 6.75in
\oddsidemargin -0.15in
\evensidemargin -0.15in
\textheight 9in
\topmargin -0.5in
%%%%%%%%%%%%%%%% end my additions to header

\usepackage[T1]{fontenc}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \usepackage{fontspec}
  \ifxetex
    \usepackage{xltxtra,xunicode}
  \fi
  \defaultfontfeatures{Mapping=tex-text,Scale=MatchLowercase}
  \newcommand{\euro}{â‚¬}
\fi
% use microtype if available
\IfFileExists{microtype.sty}{\usepackage{microtype}}{}
\ifxetex
  \usepackage[setpagesize=false, % page size defined by xetex
              unicode=false, % unicode breaks when used with xetex
              xetex]{hyperref}
\else
  \usepackage[unicode=true]{hyperref}
\fi
\hypersetup{breaklinks=true,
            bookmarks=true,
            pdfauthor={},
            pdftitle={rgbif: a package for working with species occurrence data in R},
            colorlinks=true,
            urlcolor=blue,
            linkcolor=magenta,
            pdfborder={0 0 0}}
\urlstyle{same}  % don't use monospace font for urls
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\setcounter{secnumdepth}{0}
% Pandoc toggle for numbering sections (defaults to be off)
\setcounter{secnumdepth}{0}
% Pandoc header



\begin{document}
\begin{frontmatter}

  \title{rgbif: a package for working with species occurrence data in R}
    \author[cstar]{Scott Chamberlain\corref{c1}}
   \ead{scott(at)ropensci.org} 
   \cortext[c1]{Corresponding author}
      \address[cstar]{University of California, Berkeley, CA, USA}    
  
  \begin{abstract}
  \begin{enumerate}
  \def\labelenumi{\arabic{enumi}.}
  \item
    xxx
  \item
    xxx
  \item
    xxx
  \item
    xxxx
  \end{enumerate}
  \end{abstract}
  
 \end{frontmatter}


\section{Introduction}\label{introduction}

Users of the popular statistical and mathematical computing platform R
(R Core Team 2014) enjoy a wealth of readily installable comparative
phylogenetic methods and tools (O'Meara 2014). Exploiting the
opportunities arising from this wealth for complex and integrative
comparative research questions relies on the ability to reuse and
integrate previously generated or published data and metadata. The
expanding data exchange needs of the evolutionary research community are
rapidly outpacing the capabilities of most current and widely used data
exchange standards (Vos \emph{et al.} 2012), which were all developed a
decade or more ago. This has resulted in a radiation of different data
representations and exchange standard ``flavors'' that are no longer
interoperable at the very time when the growth of available data and
methods has made that interoperability most valuable. In response to the
unmet needs for standardized data exchange in phylogenetics, a modern
XML-based exchange standard, called NeXML, has recently been developed
(Vos \emph{et al.} 2012). NeXML comprehensively supports current data
exchange needs, is predictably machine-readable, and is forward
compatible.

The exchange problem for phylogenetic data is particularly acute in
light of the challenges in finding and sharing phylogenetic data without
the otherwise common loss of most data and metadata semantics (Stoltzfus
\emph{et al.} 2012; Drew \emph{et al.} 2013; Cranston \emph{et al.}
2014). For example, the still popular NEXUS file format (Maddison
\emph{et al.} 1997) cannot consistently represent horizontal gene
transfer or ambiguity in reading a character (such as a DNA sequence
base pair). This and other limitations have led to modifications of
NEXUS in different ways for different needs, with the unfortunate result
that NEXUS files generated by one program can be incompatible with
another (Vos \emph{et al.} 2012). Without a formal grammar, software
based on NEXUS files may also make inconsistent assumptions about
tokens, quoting, or element lengths. Vos et al. (2012) estimates that as
many as 15\% of the NEXUS files in the CIPRES portal contain
unrecoverable but hard to diagnose errors.

A detailed account of how the NeXML standard addresses these and other
relevant challenges can be found in Vos et al. (2012). In brief, NeXML
was designed with the following important properties. First, NeXML is
defined by a precise grammar that can be programmatically
\textbf{validated}; i.e., it can be verified whether a file precisely
follows this grammar, and therefore whether it can be read (parsed)
without errors by software that uses the NeXML grammar (e.g.~RNeXML) is
predictable. Second, NeXML is \textbf{extensible}: a user can define
representations of new, previously unanticipated information (as we will
illustrate) without violating its defining grammar. Third and most
importantly, NeXML is rich in \textbf{computable semantics}: it is
designed for expressing metadata such that machines can understand their
meaning and make inferences from it. For example, OTUs in a tree or
character matrix for frog species can be linked to concepts in a
formally defined hierarchy of taxonomic concepts such as the Vertebrate
Taxonomy Ontology (Midford \emph{et al.} 2013), which enables a machine
to infer that a query for amphibia is to include the frog data in what
is returned. (For a more broader discussion of the value of such
capabilities for evolutionary and biodiversity science we refer the
reader to Parr et al. (2011).)

To make the capabilities of NeXML available to R users in an easy-to-use
form, and to lower the hurdles to adoption of the standard, we present
RNeXML, an R package that aims to provide easy programmatic access to
reading and writing NeXML documents, tailored for the kinds of use-cases
that will be common for users and developers of the wealth of
evolutionary analysis methods within the R ecosystem.

\section{The rgbif package}\label{the-rgbif-package}

The \texttt{rgbif} package \ldots{}

\section{Conclusions and future
directions}\label{conclusions-and-future-directions}

\texttt{rgbif} \ldots{}

\subsection{Acknowledgements}\label{acknowledgements}

This project was supported in part by the Alfred P Sloan Foundation
(Grant 2013-6-22).

\subsection{Data Accessibility}\label{data-accessibility}

All software, scripts and data used in this paper can be found in the
permanent data archive Zenodo under the digital object identifier (DOI).
This DOI corresponds to a snapshot of the GitHub repository at
\href{https://github.com/ropensci/rgbif}{github.com/ropensci/rgbif}.

\section*{References}\label{references}
\addcontentsline{toc}{section}{References}

Cranston, K., Harmon, L.J., O'Leary, M.A. \& Lisle, C. (2014). Best
practices for data sharing in phylogenetic research. \emph{PLoS Curr}.
Retrieved from
\url{http://dx.doi.org/10.1371/currents.tol.bf01eff4a6b60ca4825c69293dc59645}

Drew, B.T., Gazis, R., Cabezas, P., Swithers, K.S., Deng, J., Rodriguez,
R., Katz, L.A., Crandall, K.A., Hibbett, D.S. \& Soltis, D.E. (2013).
Lost branches on the tree of life. \emph{PLoS Biol}, \textbf{11},
e1001636. Retrieved from
\url{http://dx.doi.org/10.1371/journal.pbio.1001636}

Maddison, D., Swofford, D. \& Maddison, W. (1997). NEXUS: An extensible
file format for systematic information. \emph{Syst. Biol.}, \textbf{46},
590--621. Retrieved from
\url{http://www.ncbi.nlm.nih.gov/pubmed/11975335}

Midford, P., Dececchi, T., Balhoff, J., Dahdul, W., Ibrahim, N., Lapp,
H., Lundberg, J., Mabee, P., Sereno, P., Westerfield, M., Vision, T. \&
Blackburn, D. (2013). The vertebrate taxonomy ontology: A framework for
reasoning across model organism and species phenotypes. \emph{J. Biomed.
Semantics}, \textbf{4}, 34. Retrieved from
\url{http://dx.doi.org/10.1186/2041-1480-4-34}

O'Meara, B. (2014). CRAN task view: Phylogenetics, especially
comparative methods. Retrieved from
\url{http://cran.r-project.org/web/views/Phylogenetics.html}

Parr, C.S., Guralnick, R., Cellinese, N. \& Page, R.D.M. (2011).
Evolutionary informatics: unifying knowledge about the diversity of
life. \emph{Trends in ecology \& evolution}, \textbf{27}, 94--103.
Retrieved from \url{http://www.ncbi.nlm.nih.gov/pubmed/22154516}

R Core Team. (2014). \emph{R: A language and environment for statistical
computing}. R Foundation for Statistical Computing, Vienna, Austria.
Retrieved from \url{http://www.R-project.org/}

Stoltzfus, A., O'Meara, B., Whitacre, J., Mounce, R., Gillespie, E.L.,
Kumar, S., Rosauer, D.F. \& Vos, R.A. (2012). Sharing and re-use of
phylogenetic trees (and associated data) to facilitate synthesis.
\emph{BMC Research Notes}, \textbf{5}, 574. Retrieved from
\url{http://dx.doi.org/10.1186/1756-0500-5-574}

Vos, R.A., Balhoff, J.P., Caravas, J.A., Holder, M.T., Lapp, H.,
Maddison, W.P., Midford, P.E., Priyam, A., Sukumaran, J., Xia, X. \&
Stoltzfus, A. (2012). NeXML: Rich, extensible, and verifiable
representation of comparative data and metadata. \emph{Systematic
Biology}, \textbf{61}, 675--689. Retrieved from
\url{http://dx.doi.org/10.1093/sysbio/sys025}

\end{document}


